java/vector/src/main/codegen/templates/MapWriters.java

```diff
@@ -102,7 +116,7 @@ public Field getField() {
 
   @Override
   public MapWriter map(String name) {
-      FieldWriter writer = fields.get(name.toLowerCase());
+      FieldWriter writer = fields.get(handleCase(name));
```

Comment:
@alphalfalfa he's referring to the changes in this function and below. I think Jacques is proposing having a `CaseSensitiveMapWriter`